docs/DeveloperGuide.md

```diff
@@ -1,34 +1,662 @@
+
+![](hrDiagramPic/HrArchitecture.png)
+
+There are a total of 7 commands under HR feature.
+ `CommandAddMember`, `CommandDelMember`, `CommandViewMember`  ,`CommandListConnection`, `CommandSearchMember` , `CommandListProfAdmin` and `CommandChangeMemberInfo`. 
+ 
+The implementation for each command is described in detail below.
+
+**3.5.1. Add/delete member feature**  
+3.5.1.1. Current Implementation  
+The add/delete member mechanism is facilitated by `CommandAddMember` and `CommandDelMember` classes. The 
+`CommandAddMember` class in `seedu.duke.hr` handles adding members. It adds a new `Member` instance according to 
+`userInput` into `MemberList`.  
+The `CommandDelMember` class in the same package handles deleting members. It deletes a certain `Member` instance 
+according to the index provided by `userInput` from `MemberList`.  
+These two classes implement the following operations:  
+* `CommandAddMember#execute()` - Adds a new member into the `MemberList` according to `userInput`.  
+* `CommandDelMember#execute()` - Deletes a certain member from `MemberList` according to the index provided by 
+`userInput`.  
+
+Given below is an example usage scenario and how the add/delete member behaves at each step.  
+
+Step 1. The user launches the application for the first time. The `MemberList` will be initialized with no `Member` in 
+it.  
+
+![](hrDiagramPic/2-1S1.png)
+
+Step 2. The user executes `hr addMember /n john sterling /p 12345678 /e 123@gmail.com /r member` command to add a member
+ with name "John Sterling", phone number "12345678", email "123@gmail.com" and role "member" into member list. The 
+ `hr addMember` command calls `CommandAddMember#execute()`, which then calls `MemberList#findByName()`, 
+ `MemberList#standardizeMemberName()` and `MemberList#addToList()`. `MemberList#findByName()` finds the `Member` in the 
+ list by the given member name. `MemberList#standardizeMemberName()` standardize the member name input by the user by 
+ capitalizing the first letter in each word in the name. Then, `MemberList#addToList()` adds a `Member` with its 
+ `memberName` as `John Sterling`, `memberPhone` as `12345678`, `memberEmail` as `123@gmail.com`, and `memberRole` as 
+ `member` into `MemberList`. 
+
+The following shortcut commands can achieve the same result: <br/>
+`hr add /n john sterling /p 12345678 /e 123@gmail.com /r member`<br/>
+`hr a /n john sterling /p 12345678 /e 123@gmail.com /r member`<br/>
+
+![](hrDiagramPic/2-1S2.png)
+
+Step 3. The user executes `hr delMember 1` command to delete the member in the member list. The `hr delMember`
+command calls `CommandDelMember#execute()`, causing the `Member` of index 1 removed from `MemberList`.  
+
+ The following shortcut commands can achieve the same result: <br/>
+ `hr delete 1`<br/>
+ `hr d 1`<br/>
+
+![](hrDiagramPic/2-1S3.png)
+
+The sequence diagram for adding a member is as shown below:
+
+![CommandAddMember](hrDiagramPic/CommandAddMember.png)
```

Comment:
![image](https://user-images.githubusercontent.com/60458098/97661645-e516ae80-1aaf-11eb-8bde-66daf8f631ba.png)
Should there maybe be a constructor for Member here?